build: build and test examples using bazel

devversion opened the pull request:
Pushing the initial wip of building and testing the examples with Bazel. Now that we have decided on a consistent way of converting the examples to Bazel, this work can be continued now. WIP

IgorMinar commented on tools/gulp-tasks/serve.js:
```diff
@@ -23,19 +23,4 @@ module.exports = {
 
-  // Serve the examples
-  examples: (gulp) => () => {
-    const connect = require('gulp-connect');
```
can we now remove gulp-connect from package.json or do we still use it elsewhere?

devversion replied:
We still use `gulp-connect` for serving the e2e tests within `/modules`, so I think we need to keep that for now.

IgorMinar commented on packages/examples/upgrade/upgrade_example.bzl:
```diff
@@ -0,0 +1,63 @@
+  upgrade examples bootstrap their application manually, and we cannot serve all examples,
+  we need to define the devserver for each example. This macro reduces code duplication
+  for defining these targets.
+"""
```
awesome!

devversion committed:
build: switch example e2e tests to bazel

* No longer builds the example e2e tests using "tsc". The examples are now built with Bazel and can therefore be built with Ivy by using the `--define=compile=aot` switch.
* No longer runs the example e2e tests using the protractor CLI. example e2e tests are executed with the Bazel protractor rule and can therefore run incrementally.

NOTE: Unit tests found within the examples are still running within the legacy jobs.

devversion committed:
test: disable failing ivy example e2e tests

IgorMinar commented on packages/examples/forms/BUILD.bazel:
```diff
@@ -0,0 +1,59 @@
+        ["**/*.ts"],
+        exclude = ["**/*_spec.ts"],
+    ),
+    # TODO: FW-1004 Type checking is currently not complete.
```
I worry that we'll forget to turn this on later. Can you create a sub task to undo this under fw-1004 so that we don't forget

IgorMinar replied:
And list the disabled targets

devversion replied:
Done. See FW-1005

IgorMinar reviewed (approved):
Great work
